gobblin-core/src/main/java/gobblin/source/extractor/hadoop/HadoopFsHelper.java

```diff
@@ -0,0 +1,95 @@
+package gobblin.source.extractor.hadoop;
+
+import gobblin.configuration.ConfigurationKeys;
+import gobblin.configuration.State;
+import gobblin.source.extractor.filebased.FileBasedHelper;
+import gobblin.source.extractor.filebased.FileBasedHelperException;
+import gobblin.util.ProxiedFileSystemWrapper;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.List;
+
+public abstract class HadoopFsHelper implements FileBasedHelper {
+  private final State state;
+  private final Configuration configuration;
+  private FileSystem fs;
+
+  protected HadoopFsHelper(State state, Configuration configuration) {
+    this.state = state;
+    this.configuration = configuration;
+  }
+
+  protected State getState() {
+    return this.state;
+  }
+
+  public FileSystem getFileSystem() {
```

Comment:
`HadoopFsHelper` needs to be public because people implementing other FsHelper outside of the gobblin project will need access.  `getFileSystem` needs to be public because it is used by `SizeAwareFileBasedHelperDecorator`.